packages/firebase_database/lib/src/database_reference.dart

```diff
@@ -162,12 +162,13 @@ class DatabaseReference extends Query {
       'path': path,
       'transactionKey': transactionKey,
       'transactionTimeout': timeout.inMilliseconds
-    }).then((Map<String, dynamic> result) {
+    }).then((dynamic response) {
+      final Map<String, dynamic> message = response;
```

Comment:
The Map describes a transaction result, not a message. Maybe refactor to a
```dart
TransactionResult toTransactionResult(dynamic response) { ... }
```
Then the name of the local variable can be just `map` or similar, because the function name already documents what we are doing.

Reply:
done